hazelcast/src/test/java/com/hazelcast/nio/tcp/OperationPacketFilter.java

```diff
@@ -41,7 +42,7 @@ public final boolean allow(Packet packet, Address endpoint) {
 
     private boolean allowOperation(Packet packet, Address endpoint) {
         try {
-            ObjectDataInput input = serializationService.createObjectDataInput(packet);
+            ObjectDataInput input = serializationService.createObjectDataInput(new HeapData(packet.getPayload()));
```

Comment:
We are just extracting a few fields from the Packet here, don't we? Maybe we can skip the whole serialization part if the Packet would have `isIdentified()`, `isCompressed()`, `getFactoryId()` and `getType()` methods, which would directly peak into its bytes. This would also get rid of the introduced `new HeapData()`.